zeppelin-plugins/launcher/k8s-standard/src/main/java/org/apache/zeppelin/interpreter/launcher/K8sStandardInterpreterLauncher.java

```diff
@@ -95,9 +100,11 @@ String getHostname() {
    */
   private String getZeppelinService(InterpreterLaunchContext context) throws IOException {
     if (isRunningOnKubernetes()) {
+      //The namespace of zeppelin server can only be read from Config.KUBERNETES_NAMESPACE_PATH while it runs in k8s cluster, it may be different from the namespace of interpreter
+      String serverNamespace = readFile(Config.KUBERNETES_NAMESPACE_PATH, Charset.defaultCharset()).trim();
```

Comment:
Please move this part to a separate function. Something like `getServerNamespace()`.